client/k8s/k8s.go

```diff
@@ -0,0 +1,95 @@
+package k8s
+
+import (
+	"net/url"
+
+	microerror "github.com/giantswarm/microkit/error"
+	micrologger "github.com/giantswarm/microkit/logger"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	// Maximum QPS to the master from this client.
+	MaxQPS = 100
+	// Maximum burst for throttle.
+	MaxBurst = 100
+)
+
+// TLSClientConfig contains settings to enable transport layer security.
+type TLSClientConfig struct {
+	CAFile  string
+	CrtFile string
+	KeyFile string
+}
+
+// Config contains the common attributes to create a Kubernetes Clientset.
+type Config struct {
+	// Dependencies.
+	Logger micrologger.Logger
+
+	// Settings
+	Address   string
+	InCluster bool
+	TLS       TLSClientConfig
+}
+
+func newRawClientConfig(config Config) *rest.Config {
+	tlsClientConfig := rest.TLSClientConfig{
+		CertFile: config.TLS.CrtFile,
+		KeyFile:  config.TLS.KeyFile,
+		CAFile:   config.TLS.CAFile,
+	}
+	rawClientConfig := &rest.Config{
+		Host:            config.Address,
+		QPS:             MaxQPS,
+		Burst:           MaxBurst,
+		TLSClientConfig: tlsClientConfig,
+	}
+
+	return rawClientConfig
+}
+
+func getRawClientConfig(config Config) (*rest.Config, error) {
+	var rawClientConfig *rest.Config
+	var err error
+
+	if config.InCluster {
+		config.Logger.Log("debug", "creating in-cluster config")
+		rawClientConfig, err = rest.InClusterConfig()
+		if err != nil {
+			return nil, microerror.MaskAny(err)
+		}
```

Comment:
Maybe a nitpicking, but may be replaced with `return rawClientConfig, microerror.MaskAny(err)`. Then `else` is not needed.